File: src/main/java/com/cruftlab/words/service/WordService.java
package com.cruftlab.words.service;

import com.cruftlab.words.model.Word;
import com.cruftlab.words.repository.WordRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

/**
 * Service wrapper for {@link WordRepository}
 */
@Service
public class WordService {
    private final WordRepository wordRepository;

    @Autowired
    public WordService(WordRepository wordRepository) {
        this.wordRepository = wordRepository;
    }

    public Iterable<Word> findAll() {
        return wordRepository.findAll();
    }

    public Iterable<Word> findRandom(int numberOfWords) {
        return wordRepository.findRandom(numberOfWords);
    }
}

File: src/main/kotlin/com/cruftlab/words/service/WordService.kt
package com.cruftlab.words.service

import com.cruftlab.words.model.Word
import com.cruftlab.words.repository.WordRepository
import org.springframework.stereotype.Service

/**
 * Service wrapper for {@link WordRepository}
 */
@Service
class WordService(val wordRepository: WordRepository) {
    fun findAll() : Iterable<Word> {
        return wordRepository.findAll()
    }

    fun findRandom(numberOfWords: Int) : Iterable<Word> {
        return wordRepository.findRandom(numberOfWords)
    }
}
